### PR TITLE
chore: add first-time-dialog test for electron

### PR DESCRIPTION
**Pull request by karanbirsingh**
#### Description of changes

This PR follows up on #2327 by adding an a11y end-to-end test for the first-time-dialog in unified. It's helpful to have the e2e test for both web & unified because of layering scenarios that don't show up in the web test, such as the issue in https://github.com/microsoft/accessibility-insights-web/issues/2316

alternative implementations:
- fully fold the `first-time-dialog` into `DeviceConnectionDialogController` by adding an explicit `dismissTelemetryDialog()` method. Remove the existing `suppressFirstTimeDialog` flag in `createApplication`. This would collapse this new e2e test file with the existing `device-connection-dialog` tests, reducing the number of times we need to restart the app
- move `this.waitForSelector(CommonSelectors.rootContainer)` into `AppController` and similarly refactor it and/or `scanForAccessibilityIssues` so that this test file doesn't have to touch `DeviceConnectionDialogController` at all 

#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [ ] Added/updated relevant unit test(s) (and ran `yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

**Comment by waabid** [comment]
> such as the issue in #2374 

This points to a dependabot PR that doesn't seem to have any issues?

File: src/tests/electron/tests/first-time-dialog.test.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    let app: AppController;
+    let dialog: DeviceConnectionDialogController;
+
+    beforeEach(async () => {
+        app = await createApplication({ suppressFirstTimeDialog: false });
+        dialog = await app.openDeviceConnectionDialog();
+    });
+
+    afterEach(async () => {
+        dialog = null;
```
Comment (waabid) [inline src/tests/electron/tests/first-time-dialog.test.ts:18]: This line doesn't seem necessary with dialog being set every time in beforeEach but I might be missing something here ...

Comment (karanbirsingh) [reply src/tests/electron/tests/first-time-dialog.test.ts:18]: I saw this in the other e2e test setups and assumed it helped cleanly end the webdriver process, but I can't find a reference now. I'll merge for now & check with the team

**Comment by karanbirsingh** [comment]
> > such as the issue in #2374
> 
> This points to a dependabot PR that doesn't seem to have any issues?

Thanks, I meant #2316

**Pull request merged**